src/unix/core.c

```diff
@@ -983,3 +984,53 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_os_homedir(char* buffer, size_t* size) {
+  struct passwd* pw;
+  char* buf;
+  size_t len;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  /* Check if the HOME environment variable is set first */
+  buf = getenv("HOME");
+
+  if (buf != NULL) {
+    len = strlen(buf);
+
+    if (len >= *size) {
+      *size = len;
+      return -ENOBUFS;
+    }
+
+    memcpy(buffer, buf, len + 1);
+    *size = len;
+
+    return 0;
+  }
+
+  /* HOME is not set, so call getpwuid() */
+  errno = 0;
+  pw = getpwuid(getuid());
```

Comment:
I'd rather you stick to `getpwuid_r()` here.  `getenv()` at least is MT-safe on most platforms if no other threads concurrently modify the environment, but `getpwuid()` is basically always unsafe.